src/test/java/sign/Ed25519Test.java

```diff
@@ -40,6 +42,20 @@ public void testKeyPairFromSeed() {
         Assertions.assertNotNull(myKeyPair.getPublicKey());
     }
 
+    @Test
+    public void testKeyPairFromSeedAndNonce() {
+        byte[] seed = new byte[]{-72, -39, -90, -96, 104, -56, -55, -33, -112, 4, -57, 50, -99, 55, -72, -116, 102, -113, -39, -88, -48, -103, -34, -60, 76, -51, -78, 92, 32, -53, -46, 115};
+
+        KeyPair myKeyPair = ed25519.keyPairFromSeed(seed);
+
+        Assertions.assertArrayEquals(
+                myKeyPair.getPrivateKey().getBytes(),
+                new byte[]{83, -69, -105, 5, 58, 122, -47, -83, 63, 15, -105, -56, -117, 48, 88, 79, 96, -102, 119, 47, -42, -40, 43, 110, -124, -38, 105, 12, -2, -54, -55, -94, -64, -54, 94, -105, 47, 81, 79, 39, 31, -38, -89, 12, -104, -96, -40, 86, 9, -76, 100, -56, 86, 33, 29, -105, -112, 29, 102, -4, 77, 120, 57, -47});
+        Assertions.assertArrayEquals(
+                myKeyPair.getPublicKey().getBytes(),
+                new byte[]{-64, -54, 94, -105, 47, 81, 79, 39, 31, -38, -89, 12, -104, -96, -40, 86, 9, -76, 100, -56, 86, 33, 29, -105, -112, 29, 102, -4, 77, 120, 57, -47});
+    }
+
```

Comment:
what about fail scenarios?

Reply:
Here I am testing determinism (there are other tests to check if it is properly working + failing scenarios), those byte arrays are completely arbitrary.
Moreover, every single byte array will give you a key pair.

Given those, I do not see the point of testing failing scenario - if this test provides this outcome it means it is deterministic, which includes the function not being non-deterministic.